Content.Server/_Moffstation/GameTicking/Rules/GunGameRuleSystem.cs

```diff
@@ -193,59 +190,52 @@ private void ProgressPlayerReward(GunGamePlayerTrackingInfo playerInfo, GunGameR
-    private void SpawnCurrentWeapon(GunGamePlayerTrackingInfo playerInfo, GunGameRuleComponent rule)
+    private void RefreshPlayerLoadout(GunGamePlayerTrackingInfo playerInfo, GunGameRuleComponent rule)
     {
         if (!_mind.TryGetMind(playerInfo.UserId, out _, out var mind))
             return;
 
         if (mind.OwnedEntity is not { } playerEntity)
             return;
 
-        var gear = EnsureComp<GunGameTrackerComponent>(playerEntity);
-        DeleteCurrentWeapons((playerEntity, gear), rule);
+        var currentGear = EnsureComp<GunGameTrackerComponent>(playerEntity);
+        DeleteCurrentLoadout((playerEntity, currentGear), rule);
 
         if (playerInfo.RewardQueue.Count == 0) // If we somehow try to spawn somebody's weapon after they win
             return;
 
-        var slotTryOrder = new Queue<string>(rule.SlotTryOrder);
-
         // Peek the player's queue, and spawn their items
-        foreach (var item in _entityTables.GetSpawns(playerInfo.RewardQueue.Peek()))
-        {
-            SpawnItemAndEquip(
-                (playerEntity, gear),
-                item,
-                slotTryOrder,
-                rule);
-        }
+        SpawnPlayerLoadout((playerEntity, currentGear), playerInfo.RewardQueue.Peek(), rule);
     }
 
     /// <summary>
-    /// Attempts to spawn an item on a player in either their hands or the next available equipment slot according to
-    /// <see cref="Components.GunGameRuleComponent.SlotTryOrder"/>.
+    /// Spawns all the items specified in the player's current loadout. Dropping anything in a slot being filled.
+    /// This is used instead of one of the pre-existing equipping methods since we need to keep track of what is being
+    /// spawned.
     /// </summary>
-    private void SpawnItemAndEquip(Entity<GunGameTrackerComponent> player,
-        EntProtoId itemProto,
-        Queue<string> slotTryOrder,
-        GunGameRuleComponent rule)
+    private void SpawnPlayerLoadout(Entity<GunGameTrackerComponent> target,
+        ProtoId<StartingGearPrototype> gear,
+        GunGameRuleComponent gameRule)
     {
-        var itemEnt = Spawn(itemProto);
-        UpgradeEnergyWeapon(itemEnt, rule);
+        if (!_proto.TryIndex(gear, out var startingGear))
+            return;
 
-        player.Comp.CurrentRewards.Add(itemEnt);
-        while (slotTryOrder.TryDequeue(out var slot))
+        foreach (var (slotName, itemProto) in startingGear.Equipment)
         {
-            if (slot == "hand" && _hands.TryForcePickupAnyHand(player, itemEnt)
-                || _inventory.TryEquip(player, player, itemEnt, slot, true, true))
-                return;
+            _inventory.TryUnequip(target, slotName, true, true, false);
+            var itemEntity = Spawn(itemProto);
+            target.Comp.CurrentRewards.Add(itemEntity);
+            UpgradeEnergyWeapon(itemEntity, gameRule);
+            _inventory.TryEquip(target.Owner, itemEntity, slotName, silent: true, force: true);
```

Comment:
imo if equipping the new thing fails, the old item should remain in its slot. But that's probably not a concern for gun game.